R/kql-build.R

```diff
@@ -38,7 +38,7 @@ kql_build.op_filter <- function(op, ...)
 {
     dots <- mapply(get_expr, op$dots)
     dot_names <- mapply(all_names, dots)
-    cols <- tidyselect::vars_select(op$vars, !!! dot_names)
+    tidyselect::vars_select(op$vars, !!! dot_names)
```

Comment:
`vars_select` throws an exception if one of names in `dot_names` is not found in `op$vars`. That's where the "Unknown column `foo`" comes from.

If we removed this, a query referencing a nonexistent column would be translated to KQL and then fail on the Kusto side.

Reply:
Ok, should add a comment to that effect; otherwise it looks like a no-op.

Reply:
Alright, I'll do another commit to add a comment and while I'm at it, I'll see if it makes sense to push the environment munging code down to `partial_eval` to make it DRYer

Reply:
I added the comment, but I don't think it makes sense to alter `partial_eval` for this because `partial_eval` is recursive, and this operation shouldn't be done recursively for each symbol in the quosure, so I don't want to clutter the recursive calls with extra parameters. Instead, I extracted a named function `add_params_to_quosure` to replace the lambda I had there before. But I also don't think referencing parameters is useful in any of the other operations besides `filter` and `mutate`, so it doesn't exactly give us a lot of code reuse.